orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -685,7 +685,7 @@ class TaskController {
     pipelineExecutions.each { pipelineExecution ->
       clearTriggerStages(pipelineExecution.trigger.other) // remove from the "other" field - that is what Jackson works against
       pipelineExecution.getStages().each { stage ->
-        if (stage.context?.group) {
+        if (stage.context?.containsKey("group")) {
```

Comment:
I expected to get my mind blown and this was even better 👏

Reply:
Orca - it aims to please

Reply:
Will it be better to check & change other `stage.context.prop` type access and change if needed ?

Reply:
Line 501. also does the same stage.context.manualSkip.. ?

Reply:
> Will it be better to check & change other `stage.context.prop` type access and change if needed ?

ugh - this is virtually impossible... because we have no typed stage contexts it's impossible to know what a stage expects... but yes, it probably is almost always wrong... but it's a dangerous change.
ironically... this was also caused by same bug (https://github.com/spinnaker/orca/pull/3224)
also why i introduced https://github.com/spinnaker/orca/blob/master/orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java#L74...

Reply:
For `manualSkip` I am not actually sure what the intention is (but 80% we shouldn't be looking upstream)
But I would rather not change it because, historically, that means we break someone. While this change is only used in the retrieval codepath